server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -92,11 +101,31 @@ public Builder tokenized(boolean tokenized) {
             return super.tokenized(tokenized);
         }
 
+        public Builder ignoreMalformed(boolean ignoreMalformed) {
+            this.ignoreMalformed = ignoreMalformed;
+            return builder;
+        }
+
+        protected Explicit<Boolean> ignoreMalformed(BuilderContext context) {
+            if (ignoreMalformed != null) {
+                return new Explicit<>(ignoreMalformed, true);
+            }
+            if (context.indexSettings() != null) {
+                return new Explicit<>(IGNORE_MALFORMED_SETTING.get(context.indexSettings()), false);
+            }
+            return NumberFieldMapper.Defaults.IGNORE_MALFORMED;
```

Comment:
I'd probably introduce an own default value for BooleanFieldMapper, that way the two defaults are not tied to each other (or should they be?)

Reply:
Fixed introduced Defaults.IGNORE_MALFORMED for BooleanFieldMapper.Defaults.